src/qibo/backends/pytorch.py

```diff
@@ -239,6 +239,21 @@ def calculate_matrix_power(
         copied = super().calculate_matrix_power(copied, power, precision_singularity)
         return self.cast(copied, dtype=copied.dtype)
 
+    def calculate_jacobian_matrix(
+        self, circuit, parameters=None, initial_state=None, return_complex: bool = True
+    ):
```

Comment:
Just to make sure, this is the jacobian of the state after execution with respect to the circuit parameters, right?

Reply:
Yes. The execution per se is not strictly necessary, we do it because that's how we get the gradients. But it's jacobian of a state w.r.t. the phases of the gates that creates said state.